core/scheduler/scheduler_test.go

```diff
@@ -312,6 +347,34 @@ func TestScheduler_GetDuty(t *testing.T) {
 	require.NoError(t, sched.Run())
 }
 
+// delayer implements scheduler.delayFunc and records the deadline and returns it immediately.
+type delayer struct {
+	mu        sync.Mutex
+	deadlines map[core.Duty]time.Time
+}
+
+func (d *delayer) Get() map[core.Duty]time.Time {
```

Comment:
Is this exported method used or aimed to used outside the package?

Reply:
Also since this is a `_test` package, it is **impossible** to use it from outside the package.

Reply:
When you make a method public you are documenting that the method is to be comsumed by other packages

Reply:
1. It is impossible to import anything from a `_test.go` or `_internal_test.go` file in another package in Golang. So your second point is not valid.
2. You general approach of making _everything_ private, both types **and** fields/methods is a subjective opinion which I have already mentioned that I do not agree with. I feel making types private is sufficient to ensure encapsulation.

Reply:
> When you make a method public you are documenting that the method is to be comsumed by other packages

I do not agree with this. Only the **type** indicates whether it is used outside the package, not the methods or fields.

Reply:
> It is impossible to import anything from a _test.go or _internal_test.go file in another package in Golang. So your second point is not valid.

_internal_test.go is to test private methods, you told me so yourself

Reply:
> You general approach of making everything private, both types and fields/methods is a subjective opinion which I have already mentioned that I do not agree with. I feel making types private is sufficient to ensure encapsulation. Public fields/method then indicate that they are used by other things in the same package as opposed to only for internal use in the type itself.

public methods in private structs does not provide encapsulation in Go as shown in the code https://github.com/leolara/privateExperiment

Reply:
here there is an example of code: https://willhaley.com/blog/private-and-public-visibility-with-go-packages/

a private struct with public method or private method

Reply:
I've documented my approach here: https://github.com/corverroos/go-visibilty

Reply:
ok made it unexported.